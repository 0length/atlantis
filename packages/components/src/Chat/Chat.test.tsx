import React from "react";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import { Chat, ChatBubble } from ".";

afterEach(cleanup);

it("renders a Chat", () => {
  const tree = renderer
    .create(
      <Chat sendReplyHandler={() => alert("TODO: Implement onClick")}>
        <ChatBubble direction={"outbound"} message={"Hello bob!"} />
        <ChatBubble
          direction={"inbound"}
          message={"Oh hello jill! How are you?"}
        />
      </Chat>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
