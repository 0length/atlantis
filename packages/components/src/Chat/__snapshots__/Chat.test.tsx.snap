// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Chat 1`] = `
Array [
  <ul
    className="chat"
  >
    <li>
      <div
        className="inContainer"
      >
        <div
          className="inboundChatBubble"
        >
          Oh hello jill! How are you?
        </div>
      </div>
      <div
        className="inContainer"
      >
        <p
          className="base regular base green"
        >
          sent
        </p>
      </div>
    </li>
    <li>
      <div
        className="outContainer"
      >
        <div
          className="outboundChatBubble"
        >
          Hello bob!
        </div>
      </div>
      <div
        className="outContainer"
      >
        <p
          className="base regular base green"
        >
          sent
        </p>
      </div>
    </li>
  </ul>,
  <div>
    <div
      className="wrapper"
      style={
        Object {
          "--formField-maxLength": undefined,
        }
      }
    >
      <label
        className="label textareaLabel"
        htmlFor="123e4567-e89b-12d3-a456-426655440001"
      >
        Write a reply...
      </label>
      <textarea
        className="formField"
        id="123e4567-e89b-12d3-a456-426655440001"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
      />
    </div>
    <button
      className="button base work primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="base extraBold small uppercase white"
      >
        Send Text Message
      </span>
    </button>
  </div>,
]
`;
